src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1424,6 +1424,243 @@ public async Task TestRuntimeBaseRouteInNextLinkForPaginatedRestResponse()
+                          piecesAvailable: 100
+                          piecesRequired: 50
+                        }
+                      ) {
+                        categoryid
+                        pieceid
+                      }
+                    }";
+
+                JsonElement mutationResponse = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                    client,
+                    server.Services.GetRequiredService<RuntimeConfigProvider>(),
+                    query: graphQLMutation,
+                    queryName: "createStock",
+                    variables: null,
+                    clientRoleHeader: null
+                    );
+                Assert.IsTrue(mutationResponse.ToString().Contains("The mutation operation was successful but the current user is unauthorized to view the response due to lack of read permissions"));
+
+                // pk_query is executed in the context of Authenticated role to validate that the create mutation executed in the context of Anonymous role
+                // resulted in the creation of a new record in the database.
+                string graphQLQuery = @"
+                    {
+                      stock_by_pk(categoryid: 5001, pieceid: 5001) {
+                        categoryid
+                        pieceid
+                        categoryName
+                      }
+                    }";
+
+                string authToken = AuthTestHelper.CreateStaticWebAppsEasyAuthToken();
+
+                JsonElement queryResponse = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                            client,
+                            server.Services.GetRequiredService<RuntimeConfigProvider>(),
+                            query: graphQLQuery,
+                            queryName: "stock_by_pk",
+                            variables: null,
+                            authToken: authToken,
+                            clientRoleHeader: AuthorizationResolver.ROLE_AUTHENTICATED);
+
+                Assert.IsFalse(queryResponse.TryGetProperty("errors", out _));
+
+                // Clean-up steps. The record created by the create mutation operation is deleted to reset the database
+                // back to its original state.
+                string deleteMutation = @"
+                    mutation {
+                        deleteStock(categoryid: 5001, pieceid: 5001) {
+                        categoryid
+                        pieceid
+                        }
+                    }";
+
+                _ = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                    client,
+                    server.Services.GetRequiredService<RuntimeConfigProvider>(),
+                    query: deleteMutation,
+                    queryName: "deleteStock",
+                    variables: null,
+                    authToken: authToken,
+                    clientRoleHeader: AuthorizationResolver.ROLE_AUTHENTICATED);
+            }
```

Comment:
If this mutation fails then the database is not in an expected state if this test were to run again.

Reply:
I've added the cleanup delete mutation within a finally block to ensure that it gets executed always. 

AFAICT, the delete GraphQL mutation should succeed always as the role has the required permissions.